app/src/main/java/net/grandcentrix/blutufintegration/ui/list/ListViewModel.kt

```diff
@@ -1,29 +1,38 @@
 package net.grandcentrix.blutufintegration.ui.list
 
 import androidx.lifecycle.*
-import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.delay
 import kotlinx.coroutines.flow.*
 import kotlinx.coroutines.launch
 import net.grandcentrix.blutuf.core.Blutuf
 import net.grandcentrix.blutuf.core.api.BlutufEvent
 import net.grandcentrix.blutuf.core.api.BlutufEventResult
 import net.grandcentrix.blutufintegration.data.model.DeviceUiState
+import net.grandcentrix.blutufintegration.data.model.ProcessState
 import net.grandcentrix.blutufintegration.data.repo.BluetoothRepository
 
+private const val SCAN_TIMEOUT: Long = 5000
+
 class ListViewModel : ViewModel() {
 
-    val uiModel = BluetoothRepository.devicesStateFlow.asLiveData()
+    private val _uiModel = BluetoothRepository.devicesStateFlow.asLiveData()
+    val uiModel: LiveData<ProcessState<List<DeviceUiState>>> = _uiModel
 
-    val selectedDevice: LiveData<DeviceUiState?> = BluetoothRepository.selectedDeviceStateFlow.asLiveData()
+    private val _selectedDevice = BluetoothRepository.selectedDeviceStateFlow.asLiveData()
```

Comment:
This `MutableStateFlow` is not in the control of this viewmodel. So we do not need the two properties to hide the fact that it is mutable.
This hiding should be done in the `BluetoothRepository` itself as it has the control over it.

Reply:
yes, you’re right 👍

Reply:
In the repo class it’s not allowed to use underscore for the names (Object or top-level property name '_selectedDeviceStateFlow' should not start with an underscore)
example discussed here: https://discuss.kotlinlang.org/t/object-or-top-level-property-name-warning/6621

Reply:
yes, I forgot that you have made the repository an `object`. This is on my list that you should change that. But this is nothing for this PR.
So I would suggest that you just remove the mutable hiding here in the viewmodel and we deal with the `BluetoothRepository` later.

Reply:
👍👍